pkg/awsSGSetup/aws_security_group_setup.go

```diff
@@ -0,0 +1,320 @@
+// Copyright (c) 2020 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package awssgsetup
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"path"
+	"regexp"
+	"strings"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+)
+
+var log = logf.Log.WithName("AWS_SG_Setup")
+var TRACE = 7
+var DEBUG = 5
+
+// setupAWSSecurityGroups updates the master and worker security groups used in an
+// OpenShift AWS setup. It sets a time that should be checked before attempting
+// to do another call to setup.
+func SetupAWSSecurityGroups(ctx context.Context, client client.Client) error {
+	// Grab ConfigMap kube-system aws-creds
+	//		get aws_access_key_id and aws_secret_access_key
+	awsKeyId, awsSecret, err := getAWSCreds(ctx, client)
+	if err != nil {
+		return fmt.Errorf("failed to update AWS SecurityGroups: %s", err.Error())
```

Comment:
nitpick, you can use `%v` instead of calling `.Error()`

```suggestion
		return fmt.Errorf("failed to update AWS SecurityGroups: %v", err)
```
Any reason you've done it the way you have? It's just habit for me to use `%v`, I wonder if it's habit for you to do that^

Note that in go 1.14, you can now use `%w` to wrap the error. Which might be the correct thing to do now

Reply:
I'm not sure. I think it is habit but I've started switching to using `%v`. I did some testing after I saw you making some changes like this recently and didn't find a difference so I've started doing it the way you suggested, I just forgot here.

Reply:
Actually I had made that change to use `%v` but that was another thing I lost at some point. :angry: